node/get_price/src/index.js

```diff
@@ -0,0 +1,103 @@
+
+const getPrice= async function (req, res) {
+
+    const AlphavantageAPIkey = req.env.ALPHAVANTAGE_API_KEY;
+    const GoldAPIkey = req.env.GOLD_API_KEY; 
+    const CoinAPIkey = req.env.COIN_API_KEY;
+    const { type } = req.payload.toLowerCase();
+    let price,code;
+  
+    if (!type) {
+      throw new Error("type is required");
+    }
+  
+    if ((!AlphavantageAPIkey || AlphavantageAPIkey === "")&&(!GoldAPIkey || GoldAPIkey === "") && (!CoinAPIkey || CoinAPIkey === "")) {
+      throw new Error("API Key is required");
+    }
+
+    if (type=== "google" || type === "amazon" || type==="ibm" || type==="apple" ||type==="microsoft"||type==="tesla"||type==="meta") {
+
+      switch(type)
+      {case "google": code="GOOGL";
+      break;
+      case "amazon": code="AMZN";
+      break;
+      case "ibm": code="IBM";
+      break;
+      case "apple": code="AAPL";
+      break;
+      case "microsoft": code="MSFT";
+      break;
+      case "tesla": code="TSLA";
+      break;
+      case "meta": code="META";
+      break;
+    default:code=null}
+     fetch(`https://www.alphavantage.co/query?function=GLOBAL_QUOTE&symbol=${code}&apikey=${AlphavantageAPIkey}`)
+     .then(data=>data.json())
+     .then(data=>{
+      price=data['Global Quote']['02. open'];
+      res.staus(200).send({ "success": true, "price": price });
+     })
```

Comment:
`staus` is invalid. I also highly suggest avoiding using `then` in `async` functions because it can make the code flow a lot more difficult to follow. For example, when I tried to run this:

```shell
curl -H "X-Internal-Challenge: secret-key" \
    -H "Content-Type: application/json" \
    -X POST http://localhost:3000/ \
    -d '{
    "variables": {
        "ALPHAVANTAGE_API_KEY": "ALPHAVANTAGE_API_KEY",
        "GOLD_API_KEY": "GOLD_API_KEY",
        "COIN_API_KEY": "COIN_API_KEY"
    },
    "payload": "{\"type\": \"google\"}"
}' 
```

the command just hang because the function threw an error:

```
22:05:16 0|server | TypeError: res.staus is not a function
22:05:16 0|server |     at /usr/code-start/src/index.js:51:13
22:05:16 0|server |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

and a response was never sent back.

```suggestion
    const response = await fetch(`https://www.alphavantage.co/query?function=GLOBAL_QUOTE&symbol=${code}&apikey=${AlphavantageAPIkey}`)
    if (response.status !== 200) {
      res.json({ success: false, message: await response.text() });
      return;
    }
    const text = await response.text();
    try {
      const data = JSON.parse(text);
      const note = data['Note'];
      if (note) {
        res.json({ success: false, message: note });
        return;
      }
      const price = data['Global Quote']['02. open'];
      res.json({ "success": true, "price": price });
    } catch (e) {
      res.json({ success: false, message: text });
    }
```

Please apply my comments for the other APIs as well.